src/table/use-sticky-columns.tsx

```diff
@@ -0,0 +1,210 @@
+  styles: { [key: string]: string };
+  isSticky: boolean;
+  isLastStickyLeft: boolean;
+  isLastStickyRight: boolean;
+  isStuckToTheLeft: boolean;
+  isStuckToTheRight: boolean;
+}
+
+export function getStickyClassNames({
+  styles,
+  isSticky,
+  isLastStickyLeft,
+  isLastStickyRight,
+  isStuckToTheLeft,
+  isStuckToTheRight,
+}: GetStickyClassNamesProps) {
+  return clsx({
+    [styles['sticky-cell']]: isSticky,
+    [styles['sticky-cell-last-left']]: isStuckToTheLeft && isLastStickyLeft,
+    [styles['sticky-cell-last-right']]: isStuckToTheRight && isLastStickyRight,
+  });
+}
+
+export const useStickyColumns = ({
+  containerWidth,
+  hasSelection,
+  isWrapperScrollable,
+  stickyColumns,
+  tablePadding,
+  visibleColumnsLength,
+  wrapperRef,
+}: StickyColumnParams) => {
+  // Check if there are any sticky columns
+  const noStickyColumns = !stickyColumns?.first && !stickyColumns?.last;
+  const columnsLengthWithSelection = visibleColumnsLength + (hasSelection ? 1 : 0);
+
+  const [shouldDisable, setShouldDisable] = useState<boolean>(noStickyColumns);
+  const [tableCellRefs, setTableCellRefs] = useState<Array<React.RefObject<HTMLTableCellElement>>>([]);
+  const { cellOffsets, updateCellOffsets } = useCellOffsets(tableCellRefs);
+
+  useResizeObserver(wrapperRef, () => updateCellOffsets());
+
+  const { first = 0, last = 0 } = stickyColumns || {};
+  // Calculate the indexes of the last left and right sticky columns, taking into account the selection column
+  const lastLeftStickyColumnIndex = first + (hasSelection ? 1 : 0);
+  const lastRightStickyColumnIndex = columnsLengthWithSelection - 1 - last;
+
+  // Get the width of the first and last sticky columns using the `cellOffsets` state, or use 0 if it's not available
+  const firstStickyColumnsWidth = cellOffsets.first[lastLeftStickyColumnIndex] ?? 0;
+  const lastStickyColumnsWidth = cellOffsets.last[lastRightStickyColumnIndex] ?? 0;
+
+  // Calculate the sum of all sticky columns' widths
+  const totalStickySpace = firstStickyColumnsWidth + lastStickyColumnsWidth;
+  useEffect(() => {
+    // Check if there is enough scrollable space for sticky columns to be enabled
+    const hasEnoughScrollableSpace =
+      totalStickySpace + MINIMUM_SCROLLABLE_SPACE + tablePadding.left + tablePadding.right < (containerWidth ?? 0);
+
+    // Determine if sticky columns should be disabled based on the conditions
+    const shouldDisable = noStickyColumns || !isWrapperScrollable || !hasEnoughScrollableSpace;
+    setShouldDisable(shouldDisable);
+  }, [containerWidth, noStickyColumns, stickyColumns, totalStickySpace, tablePadding, isWrapperScrollable]);
+
+  useEffect(() => {
+    // Create new refs for the visible columns and selection column, if present
+    setTableCellRefs(tableCellRefs =>
+      [...new Array(columnsLengthWithSelection)].map(
+        (_: any, i: number) => tableCellRefs[i] || createRef<HTMLTableCellElement>()
+      )
+    );
+  }, [columnsLengthWithSelection]);
```

Comment:
What happens if I toggle the visibility of two columns, but the number of visible columns stays the same? For example, I hide one column and show another one. `columnsLengthWithSelection` would have the same value as before.

Reply:
Awesome call - had forgotten about this. I updated the dependencies 👍